cmd/kube-controller-manager/app/cloudproviders.go

```diff
@@ -32,7 +32,7 @@ func createCloudProvider(cloudProvider string, externalCloudVolumePlugin string,
 	var cloud cloudprovider.Interface
 	var loopMode ControllerLoopMode
 	var err error
-	if cloudprovider.IsExternal(cloudProvider) {
+	if cloudprovider.IsExternal(cloudProvider) || len(cloudProvider) == 0 {
```

Comment:
this doesn't feel right because `InitClouodProvider` already has handling of `""`:
https://github.com/kubernetes/kubernetes/blob/16bc6c1d02fc8e4d40ee937ce0fdf7313ce2f233/staging/src/k8s.io/cloud-provider/plugins.go#L99

Reply:
The point is `""` will go through the `IncludeCloudLoops` process.
That will trigger a series of unnecessary errors.

Reply:
as per the PR title:
> Treat empty cloud provider as external

shouldn't `IsExternal()` handle `""`?

also possibly these two cases need to change:
https://github.com/kubernetes/kubernetes/blob/16bc6c1d02fc8e4d40ee937ce0fdf7313ce2f233/staging/src/k8s.io/cloud-provider/plugins.go#L99-L107

Reply:
Thanks for your time, but `IsExternal()` only checks if the literal equals to `external`, it shouldn't do other implicit work.

Plus, both `IsExternal()` and `InitCloudProvider()` are used by kubelet/kube-apiserver/cloud-controller-manager, we should be cautious if we want to change their behaviors.

This PR just make sure `cloud-controller-manager` treat empty cloud provider as external, that will goes through the `ExternalLoops`.